lang/cs/Org.Apache.REEF.Tang/Implementations/Configuration/ConfigurationBuilderImpl.cs

```diff
@@ -85,6 +85,16 @@ protected ConfigurationBuilderImpl(IConfiguration[] confs) : this(new string[0],
         {
         }
 
+        public IConfigurationBuilder AddOptionalConfiguration(IConfiguration conf)
```

Comment:
Correct, this is what I described in the jira. What this is trying to change is number of checks we have to code on the client side.
Currently the client have to "check" if configuration need to be added 2 times: when creating IMRUJobDefinition and when ReefClient is using job definition to create job request.
If there is helper method like AddOptionalConfiguration, ReefClient could reuse it for configs which can be null. 
Just thought of alternative solution: IMRUJobDefinitionBuilder could default optional configs to predefined EMPTY configuration. This would avoid null configs and any additional handling in REEFClient. This would work for IMRU definition only.... Does that sounds like a better approach?